### PR TITLE
Rax2 help fix branch

### PR DESCRIPTION
**Pull request by vraelvrangr**
Fixes for issues brought up in issue #5399 .
If there are any other issues found with this pull request. I will be glad to fix them, just let me know.

### CODE REVIEW

**Comment by vraelvrangr** [comment]
Note: I did accidentally close this pull request, I am still quite new to using github.

**Comment by radare** [comment]
see comments and update the pr please

**Comment by radare** [comment]
also, to get green travis please rebase

File: binr/rax2/rax2.c [hunk 1]
```diff
@@ -63,6 +63,7 @@ static int format_output (char mode, const char *s) {
 }
 
 static int help () {
+	printf ("Usage: rax2 [options] [expr ...]\n");
```
Comment (radare) [inline binr/rax2/rax2.c]: this shouldnt be shown when showing the help from inside the shell of rax2 -

File: binr/rax2/rax2.c [hunk 2]
```diff
@@ -164,11 +164,14 @@ static int rax (char *str, int len, int last) {
 			return !use_stdin ();
 		return true;
 	}
-	if (!flags) {
+	if (!flags && strlen(str) == 1) {
```
Comment (radare) [inline binr/rax2/rax2.c]: strnlen is faster here

File: binr/rax2/rax2.c [hunk 3]
```diff
@@ -164,11 +164,14 @@ static int rax (char *str, int len, int last) {
...
 		if (*str == 'q')
 			return false;
-		if (*str == 'h' || *str == '?')
-			return help ();
+		if (*str == 'h' || *str == '?') {
+			help ();
+			exit (0);
```
Comment (radare) [inline binr/rax2/rax2.c]: exit is implicit in the return

**Comment by radare** [comment]
all fixed and merged

**Pull request closed without merging**